pkg/network/go/bininspect/utils.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package bininspect
+
+import (
+	"debug/dwarf"
+	"debug/elf"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/network/go/asmscan"
+	"github.com/DataDog/datadog-agent/pkg/network/go/binversion"
+	"github.com/go-delve/delve/pkg/goversion"
+)
+
+// GetArchitecture returns the `runtime.GOARCH`-compatible names of the architecture.
+// Only returns a value for supported architectures.
+func GetArchitecture(elfFile *elf.File) (GoArch, error) {
+	switch elfFile.FileHeader.Machine {
+	case elf.EM_X86_64:
+		return GoArchX86_64, nil
+	case elf.EM_AARCH64:
+		return GoArchARM64, nil
+	}
+
+	return "", fmt.Errorf("unsupported architecture")
+}
+
+// HasDwarfInfo attempts to parse the DWARF data and look for any records.
+// If it cannot be parsed or if there are no DWARF info records,
+// then it assumes that the binary has been stripped.
+func HasDwarfInfo(elfFile *elf.File) (*dwarf.Data, bool) {
+	dwarfData, err := elfFile.DWARF()
+	if err != nil {
+		return nil, false
+	}
+
+	infoReader := dwarfData.Reader()
+	if firstEntry, err := infoReader.Next(); err == nil && firstEntry != nil {
+		return dwarfData, true
+	}
+
+	return nil, false
+}
+
+// GetAllSymbolsByName returns all the elf file's symbols mapped by their name.
+func GetAllSymbolsByName(elfFile *elf.File) (map[string]elf.Symbol, error) {
+	regularSymbols, regularSymbolsErr := elfFile.Symbols()
```

Comment:
I thought about it again and I think we should quit only if both of the errors are not nil, the rest of the code might still work if we have only one of the two, and if not - it will fail somewhere else. I changed the err condition to 'AND' for now. What do you think?

Reply:
So thinking about this more, I think it actually might be ok if one of these fails. I assume this is being used to get all symbols and then to look for particular symbols later? In that case, if we find the symbols we're looking for even if its an incomplete list, we are good to go. Also, do we need to return all the symbols here or can we pass in a list of symbol names we are looking for and return the ones that we found?